### PR TITLE
fix inner topic protection bug

### PR DESCRIPTION
**Pull request by hangc0276**
### Changes
1. Add system topic protection for handleProduce. However, there is a bug in pulsar, i will fix it by another PR on Pulsar.
2. Fix a bug in `InnerTopicProtectionTest`. 

I turn off the system topic produce test in `InnerTopicProtectionTest` first, after fixed on Pulsar side for system topic check, and KOP merged the Pulsar master, i will turn on again.

### CODE REVIEW

**Comment by BewareMyPower** [comment]
The checkstyle plugin has been upgraded, I think it's better to rebase to master first.

**Comment by hangc0276** [comment]
> The checkstyle plugin has been upgraded, I think it's better to rebase to master first.

@BewareMyPower  It run failed on basicProduceConsume test, not check style, i rerun it first.

**Comment by hangc0276** [comment]
The related PR in Pulsar is https://github.com/apache/pulsar/pull/10529

**Comment by BewareMyPower** [comment]
Yeah, I just mean to ensure this PR could pass the latest checkstyle check.

**Comment by hangc0276** [comment]
> Yeah, I just mean to ensure this PR could pass the latest checkstyle check.

@BewareMyPower  I have rebased to master.

File: tests/src/test/java/io/streamnative/pulsar/handlers/kop/InnerTopicProtectionTest.java [hunk 1]
```diff
@@ -64,6 +64,7 @@ protected KafkaServiceConfiguration resetConfig(int brokerPort, int webPort, int
         kConfig.setBrokerDeleteInactiveTopicsEnabled(false);
         kConfig.setSystemTopicEnabled(true);
         kConfig.setTopicLevelPoliciesEnabled(true);
+        kConfig.setDefaultNumPartitions(6);
```
Comment (BewareMyPower) [inline tests/src/test/java/io/streamnative/pulsar/handlers/kop/InnerTopicProtectionTest.java:67]: Why did you set the default number of partitions here?

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java [hunk 2]
```diff
@@ -793,9 +796,16 @@ protected void handleProduceRequest(KafkaHeaderAndRequest produceHar,
...
 
-                final Consumer<PersistentTopic> persistentTopicConsumer = persistentTopic ->
-                        publishMessages(persistentTopic, byteBuf, numMessages, validRecords, fullPartitionName,
-                                offsetConsumer, errorsConsumer);
+                final Consumer<PersistentTopic> persistentTopicConsumer = persistentTopic -> {
+                    // check system topic
+                    if (persistentTopic.isSystemTopic()) {
+                        log.error("Not support produce message to system topic. topic: {}", persistentTopic);
+                        throw new InvalidTopicException(Errors.INVALID_TOPIC_EXCEPTION.message());
+                    }
```
Comment (BewareMyPower) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java:804]: It's better to put this logic o `publishMessages`. First, `publishMessages` is responsible to check if `persistentTopic` is null. 

And you shouldn't thrown exceptions here, I've already check if it's the system topic. See https://github.com/streamnative/kop/blob/0e3ef0de20f5e28c2a7adc7b6396f30f72c55920/kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java#L698-L702

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java (hunk 2 above)
Comment (hangc0276) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java]: Ok, i remove the check code.

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**